packages/react-is/src/ReactIs.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+'use strict';
+
+import {
+  REACT_ASYNC_MODE_TYPE,
+  REACT_CONTEXT_TYPE,
+  REACT_ELEMENT_TYPE,
+  REACT_FRAGMENT_TYPE,
+  REACT_PORTAL_TYPE,
+  REACT_PROVIDER_TYPE,
+  REACT_STRICT_MODE_TYPE,
+} from 'shared/ReactSymbols';
+
+// e.g. Fragments, StrictMode
+function getType(object: any) {
+  return typeof object === 'object' && object !== null ? object.type : null;
+}
+
+// e.g. Elements, Portals
+function getTypeOf(object: any) {
+  return typeof object === 'object' && object !== null ? object.$$typeof : null;
+}
+
+// e.g. Context provider and consumer
+function getTypeTypeOf(object: any) {
+  return typeof object === 'object' &&
+    object !== null &&
+    typeof object.type === 'object' &&
+    object.type !== null
+    ? object.type.$$typeof
+    : null;
+}
+
+export function typeOf(object: any) {
+  let maybeType = getType(object);
+  switch (maybeType) {
+    case REACT_ASYNC_MODE_TYPE:
+    case REACT_FRAGMENT_TYPE:
+    case REACT_STRICT_MODE_TYPE:
+      return maybeType;
```

Comment:
This seems like it would return true for `{type: 0xeacb}`. But React wouldn't consider this a fragment. It would only consider `{type: 0xeacb, $$typeof: 0xeac7}` a fragment. Same for all other things with `type`. This is a small detail but IMO we should be strict about these things.

Reply:
Yeah, that's reasonable.